test/test_um2netcdf.py

```diff
@@ -1107,3 +1117,89 @@ def test_convert_32_bit_with_float64(ua_plev_cube):
     ua_plev_cube.data = array
     um2nc.convert_32_bit(ua_plev_cube)
     assert ua_plev_cube.data.dtype == np.float32
+
+
+# fix forecast reference time tests
+@pytest.fixture
+def forecast_cube():
+    # NB: using a non-existent item code for fake forecast cube
+    return DummyCube(item_code=999)
+
+
+@pytest.fixture
+def time_points():
+    """Use for cube.coord('time').points attribute."""
+    return [-16382964.]
+
+
+@pytest.fixture
+def forecast_ref_time_coord(time_points):
+    # units & point data ripped from aiihca.paa1jan data file:
+    # cubes = iris.load("aiihca.paa1jan")
+    # cubes[0].long_name --> 'atmosphere_optical_thickness_due_to_sulphate_ambient_aerosol'
+    # cubes[0].coord("time").points --> array([-16382964.])
+    unit = cf_units.Unit(unit="hours since 1970-01-01 00:00:00")
+    assert unit.calendar == cf_units.CALENDAR_STANDARD
+
+    return iris.coords.DimCoord(time_points,
+                                standard_name=um2nc.FORECAST_REFERENCE_TIME,
+                                units=unit)
+
+
+@pytest.fixture
+def time_coord(time_points):
+    # units data ripped from aiihca data file
+    unit = cf_units.Unit(unit="hours since 1970-01-01 00:00:00",
+                         calendar=cf_units.CALENDAR_GREGORIAN)
+    assert unit.calendar == cf_units.CALENDAR_STANDARD
+
+    return iris.coords.DimCoord(time_points,
+                                standard_name=um2nc.TIME,
+                                units=unit)
+
+
+def test_fix_forecast_reference_time_exit_on_missing_ref_time(forecast_cube):
+    with pytest.raises(iris.exceptions.CoordinateNotFoundError):
+        forecast_cube.coord(um2nc.FORECAST_REFERENCE_TIME)
+
+    assert um2nc.fix_forecast_reference_time(forecast_cube) is None
```

Comment:
This is a silly test executing the exception handler block. It's missing an assertion to ensure the cube isn't modified, but it might be simpler to delete it. I can't recall why I added it...

Reply:
Ah I see. Otherwise I think leaving it in with some extra comments could be ok!

Reply:
I've added comments in 6e8b5e2f.